### DISCUSSION TITLE
Responding with streams

### DISCUSSION THREAD

**Discussion by brettwillis** [fastify/fastify#5165] [Q&A] [answered]
In some cases I want to send a streaming response, either when the response is large or generated gradually. E.g. streaming HTML, JSON, CSV etc.

According to the [docs](https://fastify.dev/docs/latest/Reference/Reply/#streams):

> send can also handle streams by setting the 'Content-Type' header to 'application/octet-stream'

Now, I think to myself, how can I respond with a stream if I want the `Content-Type` to be `application/json` or whatever? 

From testing, I don't encounter any errors when providing a stream object to `reply.send()`, but I was concerned that Fastify may be buffering the stream for parsing (e.g. reply schemas) under the hood or something like that.

However when I reviewed the code, I can't see it checking for `application/octet-stream` before allowing a stream to be sent?

https://github.com/fastify/fastify/blob/99a405a4e347629595ac5062c74d91675b1b34e5/lib/reply.js#L157-L161

Is the documentation still accurate? Can I respond with a stream with any content type?

**Comment by mcollina** [comment, marked as answer]
Sending a stream is ok even if the content type is set to json.

**Reply by brettwillis** [reply]
Ok great! I guess the documentation needs to be updated/clarified then?

Let me know if you want help with a PR.

**Reply by mcollina** [reply]
That would be amazing, thanks!

**Reply by brettwillis** [reply]
Ok see #5166
